BonusCalcApi/V1/UseCase/UpdateWeekReportsSentAtUseCase.cs

```diff
@@ -21,12 +21,17 @@ public async Task ExecuteAsync(string weekId)
         {
             var week = await _weekGateway.GetWeekAsync(weekId);
 
-            if (week is null) ThrowHelper.ThrowNotFound($"Week not found for: {weekId}");
-
-            if (week.ReportsSentAt is null)
+            if (week is null)
+            {
+                ThrowHelper.ThrowNotFound($"Week not found for: {weekId}");
+            }
+            else
             {
-                week.ReportsSentAt = DateTime.UtcNow;
-                await _dbSaver.SaveChangesAsync();
+                if (week.ReportsSentAt is null)
```

Comment:
I'm being picky, but you can reduce the indentation. You could either use `elseif`, or just add a new `if` after the first.